app/controllers/application_controller.rb

```diff
@@ -8,32 +10,32 @@ class ApplicationController < ActionController::Base
   def return_to_url
     url = session[:return_to] || root_url
     session.delete(:return_to)
-    return url
+    url
   end
 
   def url_except_param(url, param)
     uri = Addressable::URI.parse(url)
     uri.query_values = uri.query_values.except(param.to_s)
-    return uri.to_s.chomp("?")
+    uri.to_s.chomp("?")
   end
 
   def require_login
-    if !logged_in?
-      if params[:log_in_with]
-        session[:return_to] = url_except_param(request.original_url, :log_in_with)
-        logger.debug "After login will return to #{session[:return_to]}"
-        redirect_to root_path(log_in_with: params[:log_in_with])
-      else
-        redirect_to root_path
-      end
-      return
+    if logged_in?
+      @user = current_user
+    elsif params[:log_in_with]
+      login_with
+    else
+      redirect_to root_path
     end
-    @user = current_user
+  end
+
+  def login_with
+    session[:return_to] = url_except_param(request.original_url, :log_in_with)
+    logger.debug "After login will return to #{session[:return_to]}"
+    redirect_to root_path(log_in_with: params[:log_in_with])
```

Comment:
I presume you did this method extraction and not rubocop ?

If I've understood it correctly I suspect it needs making private so that it does not become usable as an action in it's own right.

Reply:
:+1: good catch, thanks

Reply:
I've fixed it, but there might be others like this, will double check myself if we go ahead with PR